### DISCUSSION TITLE
Is there a library for performing NLP on dates?

### DISCUSSION THREAD

**Discussion by deadcoder0904** [nlp-compromise/nlp_compromise#895] [Q&A] [answered]
I like [Sherlock.js](https://github.com/neilgupta/Sherlock) but it does fail a lot of times so I was wondering if `compromise` does something like that for dates?

For eg, `in 2 hours` get converted to `5th Feb 2022 2:08 pm` & `tomorrow at 9 pm` gets converted to `6th Feb 2022 9:00 pm`

Is there anything like that already available?

**Comment by spencermountain** [comment, marked as answer]
yep
https://github.com/spencermountain/compromise/tree/master/plugins/dates

http://compromise.cool/dates/

**Reply by deadcoder0904** [reply]
thanks but the demo is weird. i was wondering it would show the dates like the sherlock demo but it just shows what i entered. here i enter `1 hour` & it still shows `1 hour` & `start` & `end` don't make sense as today is `8th Feb`.

<img width="1114" alt="Screenshot 2022-02-08 at 12 24 13 PM" src="https://user-images.githubusercontent.com/16436270/152934023-67bb1f82-44dd-47f5-b707-5ec3221b5416.png">

does this work exactly like sherlock?

does it handle all internationalizations as well?

solid work bdw!

**Reply by deadcoder0904** [reply]
by weirdly i mean like `130` won't show anything but `130pm` shows `5pm`. it however shows `1:30` perfectly.

<img width="1156" alt="Screenshot 2022-02-08 at 5 31 36 PM" src="https://user-images.githubusercontent.com/16436270/152983154-c5c2fb84-ecbe-430e-bff4-abe5372a5969.png">

another question i have is "is there a way to get multiple outputs for one input?" like things for mac has? so when i press 12, it shows this:

![image](https://user-images.githubusercontent.com/16436270/152983576-a1c69a8b-f477-4a60-85d0-9d32bad6ae49.png)

or any way to build something like this?

**Comment by spencermountain** [comment]
130pm is a good bug. please file more if you find them.
There is no support for multiple-interpretations of ambiguous dates
cheers

**Reply by deadcoder0904** [reply]
yeah, i will if i use it 👍

i needed multiple interpretations so i could build a typeahead nlp datetimepicker. ping me if u ever build it :)
